src/detect.h

```diff
@@ -587,6 +588,11 @@ typedef struct DetectEngineCtx_ {
     ThresholdCtx ths_ctx;
 
     uint16_t mpm_matcher; /**< mpm matcher this ctx uses */
+    uint16_t spm_matcher; /**< spm matcher this ctx uses */
+
+    /* spm thread context prototype, built as spm matchers are constructed and
+     * later cloned for each thread. */
+    SpmThreadCtx *spm_thread_ctx;
```

Comment:
I think it would help me to rename this to spm_master_thread_ctx or similar. An alternative would be to use a diffirent 'SpmGlobalCtx' like construct, that would be used to create the thread context's. That is more how the Mpm variant works. Especially in the single threaded rule loading stage, it confuses a bit to already pass a 'ThreadCtx' around.